app/components/widgets/AlertBox.vue

```diff
@@ -0,0 +1,298 @@
+<template>
+<widget-editor
+  :navItems="navItems"
+  :isAlertBox="true"
+  :slots="[{ value: 'layout', label: $t('Layout') }]"
+  :selectedVariation="selectedVariation"
+>
+  <!-- Left Toolbar -->
+  <div slot="leftbar">
+    <div class="left-accordion__button">
+      <span style="text-transform: uppercase;" :class="{ active: selectedAlert === 'general' }">
+        {{ $t('Global Settings') }}
+      </span>
+    </div>
+    <div class="left-accordion__button">
+      <span class="button button--default add-alert-button" @click="toggleAddAlertMenu()">{{ $t('Add Alert') }}</span>
+      <div v-if="addAlertMenuOpen" class="add-alert-dropdown">
+        <button
+          v-for="type in alertTypes"
+          class="button button--action"
+          :key="type"
+          @click="addAlert(type)"
+        >
+          {{ $t('Add ') }}{{ alertName(type) }}
+        </button>
+      </div>
+    </div>
+    <div v-for="alert in alertTypes" v-if="wData" :key="alert" style="position: relative;">
+      <div class="left-accordion__button" :class="{ active: selectedAlert === alert }" @click="selectAlertType(alert)">
+        <i :class="{ 'icon-add': selectedAlert !== alert, 'icon-subtract': selectedAlert === alert }" />
+        <span class="left-accordion__title">{{ alertName(alert) }}</span>
+      </div>
+      <div class="left-accordion__input">
+        <validated-form  @input="save()"><toggle-input v-model="wData.settings[alert].enabled" /></validated-form>
+      </div>
+      <div
+        v-if="wData && selectedAlert === alert"
+        v-for="variation in wData.settings[alert].variations"
+        :key="variation.id"
+        @click="selectVariation(variation.id)"
+        class="variation-tile"
+        :class="{ active: selectedId === variation.id }"
+      >
+        <div class="variation-tile__image-box">
+          <img v-if="variation.settings.image.href" :src="variation.settings.image.href" />
+          <div class="variation-tile__name">
+            <input
+              type="text"
+              :value="variation.name"
+              :disabled="editingName !== variation.id"
+              :ref="`${variation.id}-name-input`"
+              @input="nameInputHandler($event.target.value)"
+              @blur="nameBlurHandler(variation.id)"
```

Comment:
This input is a deliberate antipattern to what we've been doing regarding widget inputs since there is some added functionality here that wasn't possible using our currently system, like programatically focusing the input component. Currently I have this input save on blur but I'm not sure if that should be the way to do it or a debounced on input. My worry with a debounce is they could select another input quickly and it may be overwritten since they all use the same handler.